File: 1_process1.c
#include    <stdio.h>
#include    <sys/wait.h>
#include    <unistd.h>

int main(int argc, char *argv[])
{
    pid_t pid, wpid;
    int stat;

    pid = fork();
    if (pid == 0) {
        // 子プロセスが実行する
        printf("[Child] PID:%ld PPID:%ld fork ret:%d\n", getpid(), getppid(), pid);
    } else if (pid > 0) {
        wpid = wait(&stat);
        //printf("Child exit PID:%d\n", wpid);
        // 親プロセスが実行する
        printf("[Parent] PID:%ld PPID:%ld fork ret:%d\n", getpid(), getppid(), pid);
    } else {
        printf("[Error] PID:%ld PPID:%ld fork ret:%d\n", getpid(), getppid(), pid);
    }
    // どちらも実行する
    printf("PID:%ld PPID:%ld fork ret:%d\n", getpid(), getppid(), pid);

    return 0;
}

File: 1_process1.py
# coding: utf-8
import os
#import sys

def main():
    pid = os.fork()

    if pid == 0:
        print('[Child] PID:{} PPID:{} fork ret:{}'.format(os.getpid(), os.getppid(), pid))
        #sys.stdout.flush()
        #os.execv('/bin/echo', ['/bin/echo', 'Hello Python'])
    elif pid > 0:
        os.wait()
        print('[Parent] PID:{} PPID:{} fork ret:{}'.format(os.getpid(), os.getppid(), pid))
    else:
        print('[Error] PID:{} PPID:{} fork ret:{}'.format(os.getpid(), os.getppid(), pid))

    print('PID:{} PPID:{} fork ret:{}'.format(os.getpid(), os.getppid(), pid))

if __name__ == "__main__":
    main()
